legend-engine-xt-serviceStore-executionPlan/src/main/java/org/finos/legend/engine/plan/execution/stores/service/auth/ServiceStoreConnectionProvider.java

```diff
@@ -0,0 +1,127 @@
+// Copyright 2023 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.finos.legend.engine.plan.execution.stores.service.auth;
+
+import org.apache.http.client.methods.RequestBuilder;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.HttpClients;
+import org.eclipse.collections.api.block.function.Function;
+import org.eclipse.collections.impl.utility.ListIterate;
+import org.finos.legend.authentication.credentialprovider.CredentialProviderProvider;
+import org.finos.legend.engine.connection.ConnectionProvider;
+import org.finos.legend.engine.connection.ConnectionSpecification;
+import org.finos.legend.engine.plan.execution.stores.service.IServiceStoreExecutionExtension;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.AuthenticationSchemeRequirement;
+import org.finos.legend.engine.protocol.pure.v1.model.executionPlan.nodes.SingleAuthenticationSchemeRequirement;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.authentication.specification.AuthenticationSpecification;
+import org.finos.legend.engine.protocol.pure.v1.model.packageableElement.store.service.model.SecurityScheme;
+import org.finos.legend.engine.shared.core.function.Function4;
+import org.finos.legend.engine.shared.core.identity.Credential;
+import org.finos.legend.engine.shared.core.identity.Identity;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public class ServiceStoreConnectionProvider extends ConnectionProvider<HttpConnectionBuilder>
+{
+
+    public ServiceStoreConnectionProvider(CredentialProviderProvider credentialProviderProvider)
+    {
+        super(credentialProviderProvider);
+    }
+
+    public HttpConnectionBuilder makeConnection(ConnectionSpecification connectionSpecification, AuthenticationSpecification authenticationSpecification, Identity identity) throws Exception
+    {
+        if (!(connectionSpecification instanceof ServiceStoreConnectionSpecification))
+        {
+            String message = String.format("Invalid connection specification. Expected %s but found %s", ServiceStoreConnectionSpecification.class.getCanonicalName(), connectionSpecification.getClass().getCanonicalName());
```

Comment:
hmm do we need to expsose these class names to users. I mean this is internal platform code so maybe we can throw and exception that Illegal State was encountered and ask users to reach dev team.
What do you think?

Reply:
yeah, this makes more sense.